src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx

```diff
@@ -0,0 +1,129 @@
+import { Loader } from '@openfun/cunningham-react';
+import React, { Fragment, useEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { Box, Text } from '@/components';
+import { InfiniteScroll } from '@/components/InfiniteScroll';
+
+import { TeamResponse, useTeams } from '../api/useTeams';
+import { useTeamStore } from '../store/useTeamsStore';
+
+import { PanelTeam } from './PanelTeam';
+
+type DictTeams = {
+  [page: string]: TeamResponse[];
+};
+
+export const PanelTeams = () => {
+  const { t } = useTranslation();
+
+  const [dictTeams, setDictTeams] = useState<DictTeams>({});
+  const [page, setPage] = useState(1);
+  const ordering = useTeamStore((state) => state.ordering);
+  const { data, isError, isLoading } = useTeams({
+    page,
+    ordering,
+  });
+  const containerRef = useRef<HTMLDivElement>(null);
+  const scrollLoading = useRef(false);
+  const teams = useMemo(() => {
+    let teams: TeamResponse[] = [];
+    Object.keys(dictTeams)
+      .sort()
+      .forEach((key) => (teams = teams.concat(dictTeams[key])));
+
+    return teams;
+  }, [dictTeams]);
+
+  useEffect(() => {
+    setDictTeams({});
+    setPage(1);
+    setTimeout(() => {
+      if (!containerRef.current) {
+        return;
+      }
+
+      containerRef.current.scroll({
+        top: 0,
+        behavior: 'smooth',
+      });
+    }, 50);
+  }, [ordering]);
+
+  useEffect(() => {
+    if (!data?.results) {
+      return;
+    }
+
+    setDictTeams((prev) => {
+      const newTeams = {
+        [page]: data.results,
+      };
+
+      scrollLoading.current = false;
+
+      return {
+        ...prev,
+        ...newTeams,
+      };
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data?.results]);
+
+  return (
+    <Box $height="90vh" $css="overflow: auto;" ref={containerRef}>
+      <InfiniteScroll
+        hasMore={data?.count ? data.count > teams.length : false}
```

Comment:
nit: should it work?
```suggestion
        hasMore={data?.count > teams.length}
```

Reply:
Never mind, keep your implementation if it saves time, nitpicking